src/Servers/Kestrel/Core/src/Internal/Http/ForChunkedEncodingMessageBody.cs

```diff
@@ -0,0 +1,670 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
+{
+    /// <summary>
+    ///   http://tools.ietf.org/html/rfc2616#section-3.6.1
+    /// </summary>
+    public class ForChunkedEncoding : Http1MessageBody
+    {
+        // byte consts don't have a data type annotation so we pre-cast it
+        private const byte ByteCR = (byte)'\r';
+        // "7FFFFFFF\r\n" is the largest chunk size that could be returned as an int.
+        private const int MaxChunkPrefixBytes = 10;
+
+        private long _inputLength;
+
+        private Mode _mode = Mode.Prefix;
+        private volatile bool _canceled;
+        private Task _pumpTask;
+        private Pipe _requestBodyPipe;
+        private int _userCanceled;
+
+        public ForChunkedEncoding(bool keepAlive, Http1Connection context)
+            : base(context)
+        {
+            RequestKeepAlive = keepAlive;
+
+            // For now, chunking will use the request body pipe
+            _requestBodyPipe = CreateRequestBodyPipe(context);
+            //context.InternalRequestBodyPipeReader = _requestBodyPipe.Reader;
+        }
+
+        private Pipe CreateRequestBodyPipe(Http1Connection context)
+            => new Pipe(new PipeOptions
+            (
+                pool: context.MemoryPool,
+                readerScheduler: context.ServiceContext.Scheduler,
+                writerScheduler: PipeScheduler.Inline,
+                pauseWriterThreshold: 1,
+                resumeWriterThreshold: 1,
+                useSynchronizationContext: false,
+                minimumSegmentSize: KestrelMemoryPool.MinimumSegmentSize
+            ));
+
+        private async Task PumpAsync()
```

Comment:
From here to https://github.com/aspnet/AspNetCore/pull/7603/files#diff-74fe00cb9040031782f9df9558963000R551 is unchanged from MessageBody before.